lib/logstash/filters/grok/timeout_support.rb

```diff
@@ -0,0 +1,67 @@
+# encoding: utf-8
+
+require 'timeout'
+
+# NOTE: use nested module structure so that `TimeoutError` is picked from parent (`Grok`).
+module LogStash module Filters class Grok
+  # Helper module to manage the timeout helper pieces.
+  module TimeoutSupport
+
+    def with_timeout_if(condition, context, &block)
+      if condition
+        with_timeout(context, &block)
+      else
+        yield
+      end
+    end
+
+    def with_timeout(context, &block)
```

Comment:
The boundaries of the `TimeoutSupport` module feel a bit unclear to me with the latest oop refactor. This method reaches into an ivar that it has no control over, but otherwise it is stateless.

Additionally I don't think that `the single-use handle_timeout` is really complex enough to need a separate definition; can we absorb this module's methods into the filter and simplify?

Reply:
original idea was to eventually get it to a copy-pasta state (so it can be re-used e.g. in kv)
but its far from it atm ...

Reply:
done I might tend to split out pieces even if the mixin isn't re-usable and depends on some parent state (such as an ivar in this case). so please watch me if you feel like its too much ... :lollipop: